techmatters/hrm
# CHI-1736 search filter support

## Description

Enhances the support for interpreting the filters that come from the resource service search endpoint and converting them to the desired query clauses to send to ~ElasticSearch

* Adds a map of 'filter mappings' to the ElasticSearch client search configuration - these can be configured as term filters or range filters
* Fields and their boosts are separated in the search configuration, to allow them to be adjusted or overridden easily
* If the filter specified in the API call match a configured filter, then a filter clause based on the mapping configuration will be added to the query
* If the filter is not specified by a mapping, but the value is a boolean or a number, it will be added as a term filter for a field matching the filter key
* If the filter is not specified and the value is a string or string array, it will be added to a multimatch query that searches across all the standard search fields. This query has all it's fields boosted, so matches of text specified by filters will get higher precedencwe than those matched by general text

### Checklist

### Related Issues
CHI-1736

### Verification steps

* Ensure that this version of HRM is deployed to dev
* Use cURL / Postman (or a locally modified flex build) to search specifying various filter clauses: eligibilityMinAge, eligibilityMaxAge, interpretationTranslationServicesAvailable, province, city, feeStructure, targetPopulation, howIsServiceOffered etc.
